src/java.base/share/classes/java/lang/ThreadGroup.java

```diff
@@ -792,6 +792,16 @@ private List<ThreadGroup> synchronizedSubgroups() {
         }
     }
 
+    /**
+     * Returns an array of subgroups used by JVMTI
+     */
+    private ThreadGroup[] subgroupsArray() {
+        List<ThreadGroup> groups = synchronizedSubgroups();
+        int length = groups.size();
+        return groups.toArray(new ThreadGroup[length]);
+    }
```

Comment:
Would you mind changing the comment to "Returns a snapshot of the subgroups as an array, used by JVMTI.", only to be consistent with the other methods that return subgroups as they all document it as a "snapshot". Also I think I would rename this to subgroupsAsArray as there isn't a "subgroups array".

If you want, `return groups.toArray(new ThreadGroup[0]);` will do the sizing for you.

Minor nit, you've additional blank lines after this method, I assume you only meant to add one.

Reply:
Thank you for looking at this - that's a much better name and comment and thanks for the Java hints (I copied that from somewhere else).